source/ProSymbolEditor/View Models/MilitarySymbolDockpaneViewModel.cs

```diff
@@ -181,228 +255,407 @@ public SymbolStyleItem SelectedStyleItem
+                                {
+                                    message = editOperation.ErrorMessage;
+                                }
+
+                                break;
+                            }
+                        }
+                    }
+                }
+            });
+
+            if (!creationResult)
+            {
+                MessageBox.Show(message);
             }
         }
 
-        public ICommand GoToModifyTabCommand
+        private bool CanCreatePolyFeatureFromCoordinates()
         {
-            get
+            if (GeometryType == GeometryType.Polyline)
             {
-                if (_goToModifyTabCommand == null)
+                if (PolyCoordinates.Count < 2)
                 {
-                    _goToModifyTabCommand = new RelayCommand(GoToTab, param => true);
+                    return false;
                 }
+            }
 
-                return _goToModifyTabCommand;
+            if (GeometryType == GeometryType.Polygon)
+            {
+                if (PolyCoordinates.Count < 3)
+                {
+                    return false;
+                }
+            }
+
+            if (GeometryType == GeometryType.Point)
+            {
+                return PointCoordinateValid;
+            }
+
+            foreach(CoordinateObject coordObject in PolyCoordinates)
+            {
+                if (!coordObject.IsValid)
+                {
+                    return false;
+                }
             }
+
+            return true;
         }
 
         #endregion
 
-        private async void SearchStyles(object parameter)
+        #region Event Listeners
+
+        private void OnActiveToolChanged(ArcGIS.Desktop.Framework.Events.ToolEventArgs args)
         {
-            //Make sure we have the military style file
-            if (_militaryStyleItem == null)
+            if (args.CurrentID == "ProSymbolEditor_DrawFeatureSketchTool")
             {
-                //Add military style to project
-                Task<StyleProjectItem> getMilitaryStyle = GetMilitaryStyleAsync();
-                _militaryStyleItem = await getMilitaryStyle;
+                //Toggle all down
+                AddToMapToolEnabled = true;
             }
+            else
+            {
+                //Disable all toggles
+                AddToMapToolEnabled = false;
+            }
+        }
 
-            //Clear for new search
-            if (_styleItems.Count != 0)
-                _styleItems.Clear();
-
-            //Get results and populate symbol gallery
-            IList<SymbolStyleItem> pointSymbols = await _militaryStyleItem.SearchSymbolsAsync(StyleItemType.PointSymbol, _searchString);
-            IList<SymbolStyleItem> lineSymbols = await _militaryStyleItem.SearchSymbolsAsync(StyleItemType.LineSymbol, _searchString);
-            IList<SymbolStyleItem> polygonSymbols = await _militaryStyleItem.SearchSymbolsAsync(StyleItemType.PolygonSymbol, _searchString);
-
```

Comment:
I removed the stopwatch, but the code still crashes sometimes for the first search.  It seems to be pretty hit or miss, sometimes it works, sometimes it doesn't.  Not sure why.

When the code does work, it is not usually faster.  For instance my first test with the new code came back with a time of 0.212 seconds for searching "infantry".  With the old code, it came back faster with 0.141 seconds.  And this is consistent after testing it multiple times.  So I'm not seeing much of an improvement, unless there's something I'm missing?